src/gui/attributetable/qgsfeaturefilterwidget.cpp

```diff
@@ -138,6 +140,18 @@ void QgsFeatureFilterWidget::filterVisible()
   mMainView->setFilterMode( QgsAttributeTableFilterModel::ShowVisible );
 }
 
+void QgsFeatureFilterWidget::filterInvalid()
+{
+  mFilterButton->setDefaultAction( mActionInvalidFilter );
+  mFilterButton->setPopupMode( QToolButton::InstantPopup );
+  mFilterQuery->setVisible( false );
+  mApplyFilterButton->setVisible( false );
+  mStoreFilterExpressionButton->setVisible( false );
+  const QgsExpressionContext context( QgsExpressionContextUtils::globalProjectLayerScopes( mLayer ) );
+  mMainView->filterFeatures( QStringLiteral( "is_feature_valid() = false" ), context );
+  mMainView->setFilterMode( QgsAttributeTableFilterModel::ShowInvalid );
```

Comment:
should the main view handle setting the filter in setFilterMode? It seems odd that we do this outside of that class.